submissions/AriadnaKravchuk/opp-tiny-js-world/index.js

```diff
@@ -0,0 +1,125 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/AriadnaKravchuk/a-tiny-JS-world
+   Web app: https://ariadnakravchuk.github.io/a-tiny-JS-world
+*/
+
+// ======== OBJECTS DEFINITIONS ========
+
+const SPECIES = {
+   HUMAN: "human",
+   CAT: "cat",
+   DOG: "dog"
+}
+
+const GENDER = {
+   MALE: "male",
+   FEMALE: "female"
+}
+
+const SAYING = {
+   DOG: "woof-woof!",
+   CAT: "meow!"
+}
+
+class Inhabitant {
+   constructor(species, name, gender, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.friends = [];
+      this.attributes = [];
+   }
+
+   addFriends(...friend) {
+      for (let i = 0; i < friends.length; i++) {
+         if (!this.friends.includes(friends[i])) {
+            this.friends.push(friends[i]);
+         }
+      }
+   }
+
+   printAttributes() {
+      let friendsNames = [];
+
+      for (let i = 0; i < this.friends.length; i++) {
+         friendsNames.push(this.friends[i]["name"]);
+      }
+
+      print(this.attributes.join("; ") + "; " + friendsNames.join(", "));
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(species, name, gender, saying, legs, hands) {
+      super(species, name, gender, saying);
+      this.legs = legs;
+      this.hands = hands;
+      this.attributes = [
+         this.species,
+         this.name,
+         this.gender,
+         this.saying,
+         this.legs,
+         this.hands
+      ];
```

Comment:
A child class should handle its own properties.
This is the point of encapsulation from OOP and Open/Closed principle from SOLID.
Following these principles help to create maintainable and scalable code.
Imagine, we want to rename one of the shared property.
When we have proper classes design we do this only in the class that owns it. Under current implementation we will need to change code in child classes as well.